### PR TITLE
[RW-4446][risk=no] Clean up the enableNewAccountCreation flag

### PR DESCRIPTION
**Pull request by gjuggler**
This flag was pushed to prod with last week's release (see https://github.com/all-of-us/workbench/pull/3130). We should clean this up sooner than later, since the verified institutional affiliation stuff will soon add some extra config-driven complexity to the account creation page.

---
**PR checklist**

- [X] This PR meets the Acceptance Criteria in the JIRA story
- [X] The JIRA story has been moved to Dev Review
- [X] I have run and tested this change locally

### CODE REVIEW

File: api/src/main/java/org/pmiops/workbench/api/ProfileController.java [hunk 1]
```diff
@@ -247,15 +247,25 @@ private void validateStringLength(String field, String fieldName, int max, int m
     }
   }
 
-  private void validateProfileFields(Profile profile) {
+  private void validateAndCleanProfile(Profile profile) throws BadRequestException {
```
Comment (gjuggler) [inline api/src/main/java/org/pmiops/workbench/api/ProfileController.java]: This is one minor functional change I made along the way... I felt we should probably have more of the profile object validation centralized in this method, which is called during both the createProfile and updateProfile flows.

File: api/src/main/java/org/pmiops/workbench/api/ProfileController.java [hunk 2]
```diff
@@ -247,15 +247,25 @@ private void validateStringLength(String field, String fieldName, int max, int m
...
+    if (profile.getDemographicSurvey() == null) {
+      profile.setDemographicSurvey(new DemographicSurvey());
+    }
```
Comment (freemabd) [inline api/src/main/java/org/pmiops/workbench/api/ProfileController.java]:
```suggestion
    profile.setDemographicSurvey(Optional.ofNullable(profile.getDemographicSurvey()).orElse(new DemographicSurvey()));
```

File: api/src/main/java/org/pmiops/workbench/api/ProfileController.java [hunk 3]
```diff
@@ -247,15 +247,25 @@ private void validateStringLength(String field, String fieldName, int max, int m
...
+    if (profile.getInstitutionalAffiliations() == null) {
+      profile.setInstitutionalAffiliations(new ArrayList<>());
+    }
```
Comment (freemabd) [inline api/src/main/java/org/pmiops/workbench/api/ProfileController.java]:
```suggestion
    profile.setInstitutionalAffiliations(Optional.ofNullable(profile.getInstitutionalAffiliations()).orElse(new ArrayList()));
```

File: api/src/main/java/org/pmiops/workbench/api/ProfileController.java [hunk 4]
```diff
@@ -323,28 +333,22 @@ private DbUser initializeUserIfNeeded() {
...
-    }
-    if (request.getProfile().getInstitutionalAffiliations() == null) {
-      request.getProfile().setInstitutionalAffiliations(new ArrayList<>());
+    final Profile profile = request.getProfile();
+
+    // We don't include this check in validateAndCleanProfile since some existing user profiles
+    // may have empty addresses. So we only check this on user creation, not update.
+    if (profile.getAddress() == null) {
+      throw new BadRequestException("Address must not be empty");
     }
```
Comment (freemabd) [inline api/src/main/java/org/pmiops/workbench/api/ProfileController.java]:
```suggestion
    Optional.ofNullable(profile.getAddress()).orElseThrow(() -> new BadRequestException("Address must not be empty"));
```

File: api/src/main/java/org/pmiops/workbench/api/ProfileController.java (hunk 1 above)
Comment (jmthibault79) [reply api/src/main/java/org/pmiops/workbench/api/ProfileController.java]: This seems reasonable.

File: api/src/main/java/org/pmiops/workbench/api/ProfileController.java [hunk 5]
```diff
@@ -247,15 +248,23 @@ private void validateStringLength(String field, String fieldName, int max, int m
...
 
-  private void validateProfileFields(Profile profile) {
+  private void validateAndCleanProfile(Profile profile) throws BadRequestException {
+    profile.setDemographicSurvey(
+        Optional.ofNullable(profile.getDemographicSurvey()).orElse(new DemographicSurvey()));
+    profile.setInstitutionalAffiliations(
+        Optional.ofNullable(profile.getInstitutionalAffiliations()).orElse(new ArrayList()));
+
+    String userName = profile.getUsername();
+    if (userName == null || userName.length() < 3 || userName.length() > 64) {
```
Comment (jmthibault79) [inline api/src/main/java/org/pmiops/workbench/api/ProfileController.java]: Let's make this check first.

Comment (gjuggler) [reply api/src/main/java/org/pmiops/workbench/api/ProfileController.java]: I'm not 100% sure what you meant, but I reshuffled things so the "validation" steps and the "cleanup" steps are more clearly delineated.

**Comment by gjuggler** [comment]
> There are a few more methods which can be deleted, at 228 and 372

Thanks! Removed from create-account.tsx

File: api/src/main/java/org/pmiops/workbench/api/ProfileController.java (hunk 5 above)
Comment (jmthibault79) [reply api/src/main/java/org/pmiops/workbench/api/ProfileController.java]: thanks - was just a nit, no worries

**Pull request merged**